### PR TITLE
Merge in the SINTEF fork

### PR DESCRIPTION
**Pull request by olevs**
I'm sorry for the large pull request. We have waited too long before merging in our fork. Our fork has some renaming of classes, functions and file names, so you may need to update libraries depending on OpenIGTLinkIO.

### CODE REVIEW

**Comment by lassoan** [comment]
Thank you! There are merge conflicts that would need to be resolved. Could you please rebase your branch on the latest master?

We are finalizing a new stable Plus release (just executing tests and resolving two outstanding issues), so it would be better to merge this after that release is out - hopefully in a week.

**Comment by leochan2009** [comment]
Oh, 
You guys have already added polydata device.
I just did the same stuff. 
I will review the code and fix the conflict.

File: CMakeLists.txt
Comment (leochan2009) [inline CMakeLists.txt]: set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${OpenIGTLinkIO_LIBRARY_INSTALL}")
should be:
set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${OpenIGTLinkIO_BINARY_INSTALL}")

**Review by leochan2009 — APPROVED** [review]
I made two commits:
first one is to solved the conflict in polydatadevice, also a small bug is fixed.
The second commit is to suppress warnings in the virtual functions. 
This pull request is tested with OpenIGTLinkIF module at branch: IGTLIOFullIntegration.
Polydata and Image were sent correctly.
Though due to the API change, some modification needs to be made.

One suggestion regarding the meta information. Currently equipment_uid_name, streamId and equipment_type are added to the meta section of the message. It would be great if the converter provides function to set and get meta information. Probably i would implement this after this pull request.

**Comment by olevs** [comment]
@leochan2009: That is great. Thank you for testing the changes, merging and fixing bugs. As you probably saw the code isn't necessarily finished, but we thought it was better to just merge the code than letting the forks diverge further. 

@lassoan: Are you finished with the Plus release and ready for this code to be merged in, or should we wait?

**Comment by lassoan** [comment]
Thanks to all! Plus stable will use the commit before this merge.

**Pull request merged**
